pkg/build/builder/docker.go

```diff
@@ -66,12 +66,19 @@ func (d *DockerBuilder) Build() error {
 	if err != nil {
 		return err
 	}
+	// NOTE: look here
 	sourceInfo, err := fetchSource(d.dockerClient, buildDir, d.build, d.urlTimeout, os.Stdin, d.gitClient)
 	if err != nil {
 		return err
 	}
 	if sourceInfo != nil {
-		updateBuildRevision(d.client, d.build, sourceInfo)
+		glog.V(4).Infof("Setting build revision to %#v", d.build.Spec.Revision.Git)
+		updateBuildRevision(d.build, sourceInfo)
+		if _, err = d.client.UpdateDetails(d.build); err != nil {
+			glog.V(0).Infof("error: An error occure while updating the build revision")
+			// NOTE: Handleee meee!
```

Comment:
@mfojtik was saying something about retrying at some point. that's why i left 'em like that.
do we want to retry?